transport_versioned_docs/version-1.0.0/workflow-client-server-secure.md

```diff
@@ -0,0 +1,147 @@
+* Samples root certificate, private key and certificate are provided with the sample project. Please do not use them for production. 
+* It is of the upmost importance for you to have your own certificates. The procedure to generate them using OpenSSL is detailed hereafter. 
+
+### Generating the required keys and certificates with OpenSSL
+
+It is assumed to you have openSSL installed on your machine.
+
+#### Generate the Certification Authority root certificate. 
+First thing first is to generate a private key for that will help to generate the certification authority root certificate. 
+```shell
+openssl genrsa -des3 -out clientPrivateKeyForRootCA.key 2048
+```
+You will have to provide a passphrase. Remember it as you will need it later on.
+
+Now that you have a private key, you can now generate the root certificate.
+
+```shell
+openssl req -x509 -new -nodes -key clientPrivateKeyForRootCA.key -sha256 days 1095 -out myGameClientCA.pem
+```
+You will be prompted to answer several questions. Most of the answers are not that important within the present context. 
+It is however useful to use a `common name` that makes sense for you as it is useful to identify the certificate amongst others.
+
+
+#### Generate the CA-signed certificate to use with the server
+Create now a private key for the server. 
+```shell
+openssl genrsa -out myGameServerPrivate.key 2048
+```
+From this private key, you can generate a certificate signing request (csr). 
+```shell
+openssl req -new -key myGameServerPrivate.key -out myGameServerCertificateSigningRequest.csr
+```
+You'll be prompted with the same questions as for generating the root certificate.
+The answers are no more important. Make sure to use a proper `common name` though as it might be useful in case you need to debug the certificate.
+
+Finally, using the different files generated, we can create the certificate file the server will use to authenticate itself : 
+```shell
+openssl.exe x509 -req -in myGameServerCertificateSigningRequest.csr -CA myGameClientCA.pem -CAkey clientPrivateKeyForRootCA.key -CAcreateserial -out myGameServerCertificate.crt -days 365 -sha256
+```
+You should have now generated a total of five files. Out of these, only will be used later on : 
+* The content of the `myGameClientCA.pem` file, will be used client side as the `pem` parameter.
+* On the server end, `myGameServerPrivate.key` file content will be used for the `rsaKey` parameter.
+* `myGameServerCertificate.crt` will be used by the `rsa` parameter on the server end. 
+
+### Boiler Plate file holding the secure parameters
+Create a `SecureParameters.cs` script file to hold your the certificates and the private key. Place it in the same folder as the minimal server and minimal client scripts.
+Start by enabling Unity TLS with : 
+```cs 
+#define ENABLE_MANAGED_UNITYTLS
+```
```

Comment:
Should remove that. `ENABLE_MANAGED_UNITYTLS` is automatically defined by Unity if UnityTLS supports the client API that we use for DTLS in UTP. Instead, I would list the editor versions where DTLS will work (2022.1+, 2021.2+, and some point release of 2020.3 that I'll have to dig around to find).

Reply:
Removed the Enable and added the versions. Will wait for the 2020.3 exact version to resolve this conversation.

Reply:
Earliest 2020.3 version with DTLS support is 2020.3.24f1.